File: ios/DeviceInfo.m
//
//  DeviceInfo.m
//  MyAlami
//
//  Created by Wahyu Setiawan on 16/10/22.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(DeviceInfo, NSObject)

RCT_EXTERN_METHOD(getDeviceID:(RCTResponseSenderBlock)callback)

@end

File: ios/DeviceInfo.swift
//
//  DeviceInfo.swift
//  MyAlami
//
//  Created by Wahyu Setiawan on 16/10/22.
//

import Foundation

@objc(GetUniqueId)
class GetUniqueId: NSObject {
  
  private var deviceID = "";
  
  @objc
  func getId(_ callback: RCTResponseSenderBlock) {
    if let identifierForVendor = UIDevice.current.identifierForVendor {
      deviceID = identifierForVendor.uuidString
      callback([deviceID])
    }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  func constantsToExport() -> [String: Any] {
    return ["initialDeviceID": ""]
  }
  
}
